File: J_se_les12/HW/Core/Logic/Show/ShowV1.java
package Java.J_se_les12.HW.Core.Logic.Show;

import Java.J_se_les12.HW.Core.Contact.Contact;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutErrors;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutput;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public class ShowV1 extends ShowBase {
    IConsoleOutErrors error;

    public ShowV1(Phonebook phonebook, IConsoleOutput output, IConsoleOutErrors error) {
        super(phonebook, output);
        this.error = error;
    }

    @Override
    public void showContact(int index) {
        if (this.phonebook.sizeOfBook() > 0) {
            if (index < this.phonebook.sizeOfBook()) {
                Contact contact = phonebook.getContact(index);
                output.showEntryPos(index);
                output.getfName(contact.getfName());
                output.getlName(contact.getlName());
                output.getOrganisation(contact.getOrganisation());
                output.getPhoneNumber(contact.getPhoneNumber());
                output.getNote(contact.getNote());
            } else {
                error.indexError();
            }
        } else {
            error.emptyError();
        }

    }

    public void showAll() {
        if (this.phonebook.sizeOfBook() > 0) {
            for (int i = 0; i < this.phonebook.sizeOfBook(); i++) {
                showContact(i);
            }
        } else {
            error.emptyError();
        }

    }
}

File: J_se_les13/HW/Core/Logic/Show/ShowV1.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;
using HW.Core.Contact;
namespace HW.Core.Logic.Show
{
    public class ShowV1 : ShowBase
    {
        IConsoleOutError Error { get; set; }

        public ShowV1(PhonebookV1 phonebook, IConsoleOutput output, IConsoleOutError error)
        : base(phonebook, output)
        {
            Error = error;
        }


        public override void ShowContact(int index)
        {
            if (Phonebook.SizeOfBook() > 0)
            {
                if (index < Phonebook.SizeOfBook())
                {
                    ContactV1 tmp = Phonebook.GetContact(index);
                    Output.ShowEntryPos(index);
                    Output.GetfName(tmp.GetfName());
                    Output.GetlName(tmp.GetlName());
                    Output.GetOrganisation(tmp.GetOrganisation());
                    Output.GetPhoneNumber(tmp.GetPhoneNumber());
                    Output.GetNote(tmp.GetNote());
                }
                else
                {
                    Error.IndexError();
                }
            }
            else
            {
                Error.EmptyError();
            }
        }
        public void ShowAll()
        {
            if (Phonebook.SizeOfBook() > 0)
            {
                for (int i = 0; i < Phonebook.SizeOfBook(); i++)
                {
                    ShowContact(i);
                }
            }
            else
            {
                Error.EmptyError();
            }
        }

    }
}